src/Domain/Creators/AlignedLattice.cpp

```diff
@@ -20,36 +21,58 @@ AlignedLattice<VolumeDim, TargetFrame>::AlignedLattice(
     const typename BlockBounds::type block_bounds,
     const typename IsPeriodicIn::type is_periodic_in,
     const typename InitialRefinement::type initial_refinement_levels,
-    const typename InitialGridPoints::type
-        initial_number_of_grid_points) noexcept
+    const typename InitialGridPoints::type initial_number_of_grid_points,
+    const typename BlocksToExclude::type blocks_to_exclude) noexcept
     // clang-tidy: trivially copyable
     : block_bounds_(std::move(block_bounds)),         // NOLINT
       is_periodic_in_(std::move(is_periodic_in)),     // NOLINT
       initial_refinement_levels_(                     // NOLINT
           std::move(initial_refinement_levels)),      // NOLINT
       initial_number_of_grid_points_(                 // NOLINT
           std::move(initial_number_of_grid_points)),  // NOLINT
-      number_of_blocks_by_dim_{map_array(block_bounds_, [
-      ](const std::vector<double>& v) noexcept { return v.size() - 1; })} {}
+      blocks_to_exclude_(blocks_to_exclude),          // NOLINT
```

Comment:
1. Don't NOLINT
2. `std::move`

I think as a general thing going forward we should avoid using things like `typename OPTION::type blah` in the constructors because it obscures the types. I think if the types in the constructor were `double`, `std::vector`, etc. it would be much easier to understand when to move and when not to. I think this should be a separate PR, though :)